### PR TITLE
Scala 2.12 support

### PR DESCRIPTION
**Pull request by rossabaker**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -451,11 +451,6 @@ lazy val commonSettings = Seq(
     "-Xlint:deprecation",
     "-Xlint:unchecked"
   )},
-  resolvers ++= Seq(
```
Comment (bryce-anderson) [inline build.sbt]: Since when did this happen? Nice.

File: build.sbt [hunk 2]
```diff
@@ -17,7 +17,7 @@ scalaVersion in ThisBuild := "2.11.8"
 // `set scalazVersion in ThisBuild := "7.2.4"` to change which version of scalaz
 // is used to build the project.
 scalazVersion in ThisBuild := "7.1.10"
-crossScalaVersions in ThisBuild <<= scalaVersion(Seq(_, "2.11.8", "2.12.0-RC2"))
+crossScalaVersions in ThisBuild := Seq(scalaVersion.value, "2.11.8", "2.12.0")
```
Comment (bryce-anderson) [inline build.sbt]: It would be funny to announce 2.12 support before the scala project itself.

**Comment by bryce-anderson** [comment]
It looks like 2.12 is not building correctly, so maybe my approval was premature.

**Comment by rossabaker** [comment]
I knew this would fail. I was just regenerating the checklist since we have a new target.  We will sadly be a while due to the number of dependencies.

Things like Twirl and the JSON libraries are where we would benefit from multiple repos instead of this monorepo.  I still think that's a good idea around 1.0 time.

**Comment by rossabaker** [comment]
Argonaut was released for 6.2.  I guess this is our time to jump?

**Comment by bryce-anderson** [comment]
Whats the issue with the Travis builds? It looks like the build script isn't quite right.

**Comment by rossabaker** [comment]
The new argonaut does not depend on scalaz, and I was moving too quickly.

**Comment by rossabaker** [comment]
Had to revert to Scala 2.11 for the documentation because of https://issues.scala-lang.org/browse/SI-10064.

**Pull request merged**